### DISCUSSION TITLE
How to query this api to get all the encounters for a given location in a certain game?

### DISCUSSION THREAD

**Discussion by diogofd8** [PokeAPI/pokeapi#884] [Q&A] [answered]
Hello!
I've been trying to write a simple script to take two arguments: game-version, location and to return the full list of encounters in the following format:

```
<encounter_method="" >
    <pokemon species="" rate=""/>
</encounter_method>
```

However, my experience with API is zero, I'm a mere embedded systems programmer with no experience at all with web stuff. I tried seeking on google and even chatgpt but I couldn't find any leads that would get me somewhere apart from making random api requests and hitting a wall. I tried reading the documentation but I'm a fish out of water here. Could you help me out?

**Comment by Naramsim** [comment]
The closest you can do is using graphql and filter per _generation_

```gql
query searchForPokemonInFrench {
  pokemon_v2_pokemon(
    where: {
      pokemon_v2_encounters: {
        pokemon_v2_locationarea: { name: { _eq: "sinnoh-route-229-area" } }
      }
      _and: {
        pokemon_v2_pokemonspecy: {
          pokemon_v2_generation: { name: { _eq: "generation-i" } }
        }
      }
    }
  ) {
    id
    name
  }
}
```

Our data `locations` are linked to `pokemon`, but `pokemon` are tied to `generations`, not `games`. So right now it's impossible to know if you can find a Pidgey in pokemon Blue but not in pokemon Red

**Reply by diogofd8** [reply]
I'd have to do that for every pokemon every time I want a given area/route, correct? There's no way to search for pokemon in a given location, only for location in a given pokemon. So the only way to get all the pokemon for a certain location I'd have to make n queries for n pokemon, being n the total number of pokemon in the database. Am I interpreting it correctly?

**Comment by Naramsim** [comment]
I don't understand. Can you list your inputs and your outputs clearly?

The GraphQL above lists all Pokemon belonging to a specific generation that can be found in a certain location.

Input: _generation_, _location_

Output: _pokemon_

This is the output of the query above:

```json
{
  "data": {
    "pokemon_v2_pokemon": [
      {
        "id": 16,
        "name": "pidgey"
      },
      {
        "id": 43,
        "name": "oddish"
      },
      {
        "id": 44,
        "name": "gloom"
      },
      {
        "id": 48,
        "name": "venonat"
      },
      {
        "id": 49,
        "name": "venomoth"
      },
      {
        "id": 69,
        "name": "bellsprout"
      },
      {
        "id": 70,
        "name": "weepinbell"
      },
      {
        "id": 118,
        "name": "goldeen"
      },
      {
        "id": 119,
        "name": "seaking"
      },
      {
        "id": 123,
        "name": "scyther"
      },
      {
        "id": 127,
        "name": "pinsir"
      },
      {
        "id": 129,
        "name": "magikarp"
      },
      {
        "id": 130,
        "name": "gyarados"
      }
    ]
  }
}
```

Learn more here on GraphQL: https://pokeapi.co/docs/graphql https://stackoverflow.com/a/52868182/3482533

**Comment by diogofd8** [comment, marked as answer]
Thank you for your guidance. I managed to understand the basic principle behind GraphQL and after loads of try and error I'm now somewhat able to navigate pokeAPI and get what I want from with using some coding.
